src/huggingface_hub/repository.py

```diff
@@ -393,10 +428,40 @@ def git_pull(self, rebase: Optional[bool] = False):
         except subprocess.CalledProcessError as exc:
             raise EnvironmentError(exc.stderr)
 
-    def git_add(self, pattern="."):
+    def auto_track_large_files(self, pattern="."):
+        """
+        Automatically track large files with git-lfs
+        """
+        try:
+            p = subprocess.run(
+                ["git", "ls-files", "-mo", pattern],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            )
+            files_to_be_staged = p.stdout.strip().split("\n")
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        for filename in files_to_be_staged:
+            path_to_file = os.path.join(os.getcwd(), self.local_dir, filename)
+            size_in_mb = os.path.getsize(path_to_file) / (1024 * 1024)
+
+            if size_in_mb >= 10 and not is_tracked_with_lfs(path_to_file):
```

Comment:
would be better (though way harder) to do a first pass to collect all files, then factor them into file extensions patterns... Might be overkill though. @Pierrci ?

Reply:
I feel like it would be overkill yes, who really looks at the content of a `.gitattributes` anyway? 😅

Reply:
just me probably

Reply:
but for instance if you rename a file (w/o pattern) it won't be lfs-tracked anymore

Reply:
yes, maybe remove obsolete files from the list when renamed/removed @LysandreJik? 😅 (even though I feel like 99.9999% of the time, keeping them will never present a problem)

(in case of rename, I think the current code will correctly catch the new name and add it)